src/System.Private.CoreLib/src/Microsoft/Win32/Win32Native.cs

```diff
@@ -2,185 +2,17 @@
- * execution region to prevent thread aborts while allocating your 
- * handle, and consider making your handle wrapper subclass 
- * CriticalFinalizerObject to ensure you can free the handle.  As you can 
- * probably guess, SafeHandle  will save you a lot of headaches if your code 
- * needs to be robust to thread aborts and OOM.
- *
- *
- * If you have a method that takes a native struct, you have two options for
- * declaring that struct.  You can make it a value type ('struct' in CSharp),
- * or a reference type ('class').  This choice doesn't seem very interesting, 
- * but your function prototype must use different syntax depending on your 
- * choice.  For example, if your native method is prototyped as such:
- *
- *    bool GetVersionEx(OSVERSIONINFO &amp; lposvi);
- *
- *
- * you must use EITHER THIS OR THE NEXT syntax:
- *
- *    [StructLayout(LayoutKind.Sequential, CharSet=CharSet.Auto)]
- *    internal struct OSVERSIONINFO {  ...  }
- *
- *    [DllImport(Interop.Libraries.Kernel32, CharSet=CharSet.Auto)]
- *    internal static extern bool GetVersionEx(ref OSVERSIONINFO lposvi);
- *
- * OR:
- *
- *    [StructLayout(LayoutKind.Sequential, CharSet=CharSet.Auto)]
- *    internal class OSVERSIONINFO {  ...  }
- *
- *    [DllImport(Interop.Libraries.Kernel32, CharSet=CharSet.Auto)]
- *    internal static extern bool GetVersionEx([In, Out] OSVERSIONINFO lposvi);
- *
- * Note that classes require being marked as [In, Out] while value types must
- * be passed as ref parameters.
- *
- * Also note the CharSet.Auto on GetVersionEx - while it does not take a String
- * as a parameter, the OSVERSIONINFO contains an embedded array of TCHARs, so
- * the size of the struct varies on different platforms, and there's a
- * GetVersionExA &amp; a GetVersionExW.  Also, the OSVERSIONINFO struct has a sizeof
- * field so the OS can ensure you've passed in the correctly-sized copy of an
- * OSVERSIONINFO.  You must explicitly set this using Marshal.SizeOf(Object);
- *
- * For security reasons, if you're making a P/Invoke method to a Win32 method
- * that takes an ANSI String and that String is the name of some resource you've 
- * done a security check on (such as a file name), you want to disable best fit 
- * mapping in WideCharToMultiByte.  Do this by setting BestFitMapping=false 
- * in your DllImportAttribute.
- */
+using System;
 
 namespace Microsoft.Win32
 {
-    using System;
-    using System.Runtime.InteropServices;
-    using Microsoft.Win32.SafeHandles;
-
-    /**
-     * Win32 encapsulation for System.Private.CoreLib.
-     */
+    // This class now only exists to service ilmarshalers.h, which has methods that emit
+    // calls to it (once those are cleaned up, this class can be deleted).  The DEFINE_CLASS
+    // mechanism they rely on doesn't currently support Interop.Kernel32 instead of
+    // Microsoft.Win32.Win32Native, because Interop is a class rather than a namespace.
```

Comment:
@jkotas, I ran into problems trying to delete this entirely, because ilmarshalers.h uses these two methods, and when I tried to swap in Interop.Kernel32 for Microsoft.Win32.Win32Native, it failed due to (I believe) Interop being a class rather than a namespace.  Rather than try to deal with that right now, I've left these stubs in place.  Please let me know if you're ok with that.